core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java

```diff
@@ -201,6 +252,119 @@ static ExprValue getFormattedTime(ExprValue timeExpr, ExprValue formatExpr) {
     return getFormattedString(formatExpr, TIME_HANDLERS, time);
   }
 
+  private static boolean canGetDate(TemporalAccessor ta) {
+    if (!ta.isSupported(ChronoField.YEAR)
+        || !ta.isSupported(ChronoField.MONTH_OF_YEAR)
+        || !ta.isSupported(ChronoField.DAY_OF_MONTH)) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean canGetTime(TemporalAccessor ta) {
+    if (!ta.isSupported(ChronoField.HOUR_OF_DAY)
+        || !ta.isSupported(ChronoField.MINUTE_OF_HOUR)
+        || !ta.isSupported(ChronoField.SECOND_OF_MINUTE)) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean isValidTemporalAccessor(TemporalAccessor ta) {
+    if (canGetDate(ta) || canGetTime(ta)) {
+      return true;
+    }
+    return false;
+  }
+
+  private static DateTimeFormatterBuilder getBuilderWithDefaultVals(
```

Comment:
This function is not need/used anymore

Reply:
Removed in https://github.com/Bit-Quill/opensearch-project-sql/pull/231/commits/7c1a67eaa09bcafe1a3249094055549fcb5869a9